### PR TITLE
[INLONG-7292][Sort] S3DirtySink flushes too quickly

### LINKED ISSUE #7293: [Bug] [Sort] S3DirtySink flushes too quickly
### What happened

 For jdbc multiple sink and other multiple sinks, when invoke is more frequent then the scheduler's batch interval, the dirty data will flush at a faster rate.

### What you expected to happen

let's say the batch interval is 60000. if I call dirtySink.invoke every 1 second, it will invoke, then flush 1-2 records every one second, instead of the 60s which the batch interval is supposed to be.

### How to reproduce

let's say the batch interval is 60000. if I call dirtySink.invoke every 1 second, it will invoke, then flush 1-2 records every one second, instead of the 60s which the batch interval is supposed to be.

### Environment

_No response_

### InLong version

master

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Yizhou-Yang**
### Prepare a Pull Request

- Fixes #7293
 
### Motivation

For JDBC multiple sink, the cos was filled very quickly, with a new file created every second instead of minute.

So I tried to debug, and realized that whenever dirtysink.invoke() is called, dirtysink.flush() will be called as well.

So the record will be flush into cos every second instead of every minute, and this is not how it is supposed to work.

### Modifications

I added a mechanism to prevent the flush() from being triggered in dirtysink too often, so that every two calls to dirtysink.flush() will at least have dirtysink.batchinverval seconds in between, and cos won't flush too quickly.

### CODE REVIEW

File: inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/dirty/sink/s3/S3DirtySink.java [hunk 1]
```diff
@@ -130,8 +131,16 @@ public synchronized void invoke(DirtyData<T> dirtyData) throws Exception {
     }
 
     private boolean valid() {
-        return (s3Options.getBatchSize() > 0 && size >= s3Options.getBatchSize())
-                || batchBytes >= s3Options.getMaxBatchBytes();
+        // stash dirty data for at least a minute to avoid flushing too fast
```
[the remark this answers is not in the document]
Comment (EMsnap) [reply inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/dirty/sink/s3/S3DirtySink.java:134]: and current time should be the time that calls System.currentTimeMillis(), the variable you defined here means the time before current

**Comment by healchow** [comment]
1. The `INLONG-7292` was incorrect in the PR title.
2. Please add more info for the PR.

**Review by yunqingmoswu — COMMENTED** [review]
There is already this function in the code, there is no need to add redundant and inelegant processing

**Comment by Yizhou-Yang** [comment]
> 

the batchinterval was used to make sure that it will flush every  60000ms.
But it does not stop the connector from calling .invoke before 60000ms. 
This is to stop the S3dirtysink.invoke() call from flushing before 60000ms, so it is different

**Comment by gong** [comment]
PR description shoud be format: Motivation, Modifications

**Review by EMsnap — APPROVED** [review]

**Review by gong — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by yunqingmoswu — APPROVED** [review]

**Pull request merged**